### PR TITLE
Add 5 seconds wait when retrieving distributed locks for index operations

### PR DESCRIPTION
**Pull request by papa99do**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Usability enhancement

* **What is the current behavior?** (You can also link to an open issue here)
- Error out immediately when zookeeper distributed lock cannot be retrieved for index operations. User will have a bad experience when adding docs with new fields to a new index in parallel. Most of the initial requests will error out due to lock contention. 

* **What is the new behavior (if this is a feature change)?**
- Wait up to 5 seconds when trying to retrieve the distributed lock, and check if the index still needs to be updated when the lock is required. this allows most of the initial requests to succeed with a slightly larger latency. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes

* **Related Python client changes** (link commit/PR here)
No

* **Related documentation changes** (link commit/PR here)
N/A

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/core/index_management/index_management.py [hunk 1]
```diff
@@ -198,11 +202,18 @@ def update_index(self, marqo_index: SemiStructuredMarqoIndex) -> None:
...
-            raise InternalError(f'Index {marqo_index.name} can not be updated.')
-
         with self._vespa_deployment_lock():
+            existing_index = self.get_index(marqo_index.name)
+            if not isinstance(existing_index, SemiStructuredMarqoIndex):
+                # This is just a sanity check, it should not happen since we do not expose this method to end user.
+                raise InternalError(f'Index {marqo_index.name} created by Marqo version {marqo_index.marqo_version} '
+                                    f'can not be updated.')
+
+            if (existing_index.tensor_field_map == marqo_index.tensor_field_map and
```
Comment (farshidz) [inline src/marqo/core/index_management/index_management.py]: The other thread may have added other things too, but also your field. An exact match will miss that

File: src/marqo/core/index_management/index_management.py [hunk 2]
```diff
@@ -342,8 +353,7 @@ def _vespa_deployment_lock(self):
         else:
             try:
                 with self._zookeeper_deployment_lock:
-                    logger.info(f"Retrieved the distributed lock for index operations. ")
+                    logger.debug(f"Retrieved the distributed lock for index operations. ")
                     yield
             except ZookeeperLockNotAcquiredError:
-                raise OperationConflictError("Another index creation/deletion operation is in progress. "
-                                             "Your request is rejected. Please try again later")
+                raise OperationConflictError("Your indexes are being updated. Please try again shortly.")
```
Comment (farshidz) [inline src/marqo/core/index_management/index_management.py:363]: I feel like we need a docs link here? Overall this message is confusing. Could be a todo though

Comment (papa99do) [reply src/marqo/core/index_management/index_management.py:363]: will add a todo.

File: src/marqo/core/index_management/index_management.py (hunk 1 above)
Comment (papa99do) [reply src/marqo/core/index_management/index_management.py]: Changed the logic so it only updates the index when there's fields missing.

**Review by farshidz — APPROVED** [review]

**Pull request merged**